### PR TITLE
mypy fixes in results/manager.py

### PR DESCRIPTION
**Pull request by hussain-jafari**
### Description
- *Category*: mypy
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5562

### Changes and notes
mypy fixes

### Testing
Ran mypy and pytest.

### CODE REVIEW

File: src/vivarium/framework/results/manager.py [hunk 1]
```diff
@@ -271,14 +266,22 @@ def register_binned_stratification(
...
                 raise ValueError(f"Expected a Series, but got type {type(data)}.")
-            return pd.cut(
+            data = pd.cut(
                 data, bin_edges, labels=labels, right=False, include_lowest=True, **cut_kwargs
-            )
+            )  # type: ignore [call-overload]
+            if isinstance(data, pd.Series):
+                return data
+            else:
+                raise ValueError(
```
Comment (hussain-jafari) [inline src/vivarium/framework/results/manager.py:282]: I believe the only way this could happen would be if labels were False, but it's typed and described in the doctstring as always being a list of strings.

File: pyproject.toml [hunk 2]
```diff
@@ -52,7 +52,7 @@ exclude = [
     'src/vivarium/framework/population/manager.py',
     'src/vivarium/framework/population/population_view.py',
     'src/vivarium/framework/results/interface.py',
-    'src/vivarium/framework/results/manager.py',
+    #'src/vivarium/framework/results/manager.py',
```
Comment (patricktnast) [inline pyproject.toml:55]: delete entirely

File: src/vivarium/framework/results/manager.py [hunk 3]
```diff
@@ -244,7 +239,7 @@ def register_binned_stratification(
         labels: list[str],
         excluded_categories: list[str] | None,
         target_type: str,
-        **cut_kwargs,
+        **cut_kwargs: dict[str, int | str | bool],
```
Comment (patricktnast) [inline src/vivarium/framework/results/manager.py:242]: I think you should be able to just do int | str | bool here?
https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values

Comment (hussain-jafari) [reply src/vivarium/framework/results/manager.py:242]: Yeah that works as well

File: src/vivarium/framework/results/manager.py [hunk 4]
```diff
@@ -271,14 +266,22 @@ def register_binned_stratification(
             Keyword arguments for :meth: pandas.cut.
         """
 
-        def _bin_data(data: pd.Series | pd.DataFrame) -> pd.Series:
+        def _bin_data(data: pd.Series[int | float] | pd.DataFrame) -> pd.Series[Any]:
             """Use pandas.cut to bin continuous values"""
-            data = data.squeeze()
+            # mypy wants this squeeze to return a Series or DataFrame
+            if isinstance(data, pd.DataFrame):
+                data = data.squeeze()
```
Comment (rmudambi) [inline src/vivarium/framework/results/manager.py:273]: You should do 
```python
data = data.squeeze(axis=1)
```
This guarantees you won't get a scalar.

File: src/vivarium/framework/results/observation.py [hunk 5]
```diff
@@ -163,7 +165,7 @@ def _wrap_results_gatherer(
 
     @staticmethod
     def create_empty_df(
-        requested_stratification_names: set[str],
+        requested_stratification_names: tuple[str, ...] | None,
```
Comment (rmudambi) [inline src/vivarium/framework/results/observation.py:168]: Why is this changed from a set to a tuple?

File: src/vivarium/framework/results/observation.py [hunk 6]
```diff
@@ -267,11 +269,15 @@ def create_expanded_df(
         """
 
         # Set up the complete index of all used stratifications
-        requested_and_registered_stratifications = [
-            stratification
-            for stratification in registered_stratifications
-            if stratification.name in requested_stratification_names
-        ]
+        requested_and_registered_stratifications = (
```
[the remark this answers is not in the document]
Comment (albrja) [reply src/vivarium/framework/results/observation.py:272]: + 1

File: src/vivarium/framework/results/manager.py (hunk 1 above)
Comment (albrja) [reply src/vivarium/framework/results/manager.py:282]: If labels is False doesn't that mean the user implemented something incorrectly then?

File: src/vivarium/framework/results/observation.py (hunk 5 above)
Comment (hussain-jafari) [reply src/vivarium/framework/results/observation.py:168]: I think this was originally typed incorrectly, since we actually pass in a tuple[str, ...] or None to this function.

File: src/vivarium/framework/results/manager.py (hunk 4 above)
Comment (hussain-jafari) [reply src/vivarium/framework/results/manager.py]: mypy unfortunately still complains with this approach (with the original error and a new one about passing 1 to the axis argument) so I went with a one liner instead

**Review by rmudambi — APPROVED** [review]

**Review by stevebachmeier — APPROVED** [review]
